reactor-netty-core/src/main/java/reactor/netty/transport/ClientTransportConfig.java

```diff
@@ -226,6 +230,16 @@ protected void proxyProvider(@Nullable ProxyProvider proxyProvider) {
 		}
 	}
 
+	static final ConcurrentMap<Integer, DnsAddressResolverGroup> RESOLVERS_CACHE = PlatformDependent.newConcurrentHashMap();
+
+	static DnsAddressResolverGroup getOrCreateResolver(
+			NameResolverProvider nameResolverProvider,
+			LoopResources loopResources,
+			boolean preferNative) {
+		return RESOLVERS_CACHE.computeIfAbsent(Objects.hash(nameResolverProvider, loopResources, preferNative),
```

Comment:
would it be less brittle to at least enforce `Objects.identityHashCode` on the `nameResolverProvider` and `loopResources`, just in case they get a hash code that is more prone to collisions at some point? (that would introduce a subtle bug here)

Reply:
`Objects.identityHashCode` should be `System.identityHashCode`?

Why don't we want to check the `NameResolverProvider#hashCode()`?

Reply:
hashcodes _can_ have collisions, so technically if the classe's hashcode is something that is more likely to collision than the pseudo "memory pointer" from identityHashcode, you could get the wrong resolver out of the map, couldn't you ?

Reply:
I missed the fact that `NameResolverProvider` already has a well-defined hashcode which reflects the configuration (and we want similarly configured resolvers to be shared).